internal/pkg/api/handleCheckin.go

```diff
@@ -1037,30 +1037,36 @@ func parseMeta(zlog zerolog.Logger, agent *model.Agent, req *CheckinRequest) ([]
 		return nil, nil
 	}
 
+	// Check for empty string - not valid metadata
+	if str, ok := reqLocalMeta.(string); ok && str == "" {
```

Comment:
Maybe also trim for extra safety?

```suggestion
	if str, ok := reqLocalMeta.(string); ok && strings.TrimSpace(str) == "" {
```